src/components/CreateRepoButton.js

```diff
@@ -45,7 +45,7 @@ const useStyles = makeStyles(theme => ({
 }))
 
 
-function CreateRepoButton({ active, server, owner, repo, onRefresh }) {
+function CreateRepoButton({ active, server, owner, repo, refresh, onRefresh }) {
   const {
     state: {
       authentication,
```

Comment:
I think the following piece of code in the useEffect is causing your rerender problems. 

```js
   if ( !submitCreate ) return;
```

I suggest wrapping the function in a condition instead of returning when submitCreate is false. Returning in a useEffect is reserved for cleaning up an effect (whenever the component unmounts it will execute)